csharp/src/sdk/Elements/Space.cs

```diff
@@ -0,0 +1,163 @@
+using Newtonsoft.Json;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Hypar.Geometry;
+
+namespace Hypar.Elements
+{
+    /// <summary>
+    /// A space represents an extruded boundary an occupiable region within a building.
+    /// </summary>
+    public class Space : Element, ILocateable<Polyline>, ITessellate<Mesh>, IMaterialize
+    {
+        private List<Polyline> _sides = new List<Polyline>();
+
+        /// <summary>
+        /// The bottom perimeter of the space.
+        /// </summary>
+        /// <returns></returns>
+        [JsonProperty("location")]
+        public Polyline Location{get;}
+
+        /// <summary>
+        /// The elevation of the bottom perimeter.
+        /// </summary>
+        [JsonProperty("bottom_elevation")]
+        public double BottomElevation{get;}
+
+        /// <summary>
+        /// The height of the space above its bottom elevation.
+        /// </summary>
+        [JsonProperty("height")]
+        public double Height{get;}
+
+        /// <summary>
+        /// The material of the Mass.
+        /// </summary>
+        /// <value></value>
+        [JsonIgnore]
+        public Material Material{get;set;}
+
+        /// <summary>
+        /// Construct a default space.
+        /// </summary>
+        public Space()
+        {
+            var defaultProfile = Profiles.Rectangular();
+            this.Location = defaultProfile;
+            this.BottomElevation = 0.0;
+            this.Height = 1.0;
+            this.Material = BuiltInMaterials.Default;
+        }
+
+        /// <summary>
+        /// Construct a space from a perimeter and a height.
+        /// </summary>
+        /// <param name="bottom">The bottom perimeter of the space.</param>
+        /// <param name="bottomElevation">The elevation of the bottom perimeter.</param>
+        /// <param name="height">The height of the space above the bottom elevation.</param>
+        /// <returns></returns>
+        public Space(Polyline bottom, double bottomElevation, double height)
```

Comment:
Given that we talked about having `voids`, this should be:
`public Space(Polyline perimeter IEnumerable<Polyline> voids, double elevation, double height)`

Without the `bottom` input, the `bottomElevation` makes less sense.

Reply:
Made this change. Added a TODO comment to test that voids are within the perimeter. If we implement the Geometry operations list I supplied, we could use those functions to confirm acceptable voids.